### PR TITLE
Adding Unit Tests For Primitive->ROSMessage->Primitive Translation, and misc. Primitive Cleanup

### LINKED ISSUE #135: Add unit tests for the rest of the primitives classes
**Is your feature request or enhancement related to a problem? What specifically could be better? Please describe.**
All `Primitive` classes should have unit tests.

**Describe the solution you'd like**
Add unit tests for all the following primitives. 

**Describe alternatives you've considered**
A clear and concise description of any alternative solutions or features you've considered. Since test for the `MovePrimitive` (#129 ) are already done, we can use that as a template/example. See #131 
* Catch primitive. Blocked by #125 
* Dribble primitive. Blocked by #104 
* DirectWheels primitive. Blocked by #92 
* DirectVelocity primitive. Blocked by #91 
* Stop primitive. Blocked by #90 
* MoveSpin primitive. Blocked by #89 
* Pivot primitive. Blocked by #88 
* Kick and chip primitives. Blocked by #87

### PR DESCRIPTION
**Pull request by garethellis0**
### Description
- Added some unit tests for the `Primitive` -> `ROS Message` -> `Primitive` translation for `KickPrimitive` and `ChipPrimitive`
- Refactored a few function names to be (hopefully) more representative of the functions actual functionality
- Added misc. comments in places where I was a bit confused initially

### Testing Done
Half the point of this PR is adding unit tests! :grin: 

### Resolved Issues
I think we can say that this resolves #135 (since the rest of the tests mentioned in that PR should just be added along with each primitive).

### Review Checklist
*(Please check every item to indicate your code complies with it (by changing `[ ]`->`[x]`). This will hopefully save both you and the reviewer(s) a lot of time!)*
***It is the reviewers responsibility to also make sure every item here has been covered***
- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom` 
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### Extra TODO's
- [x] Check that these changes are also applied to the `DirectVelocityPrimitive` class

***Feel free to make additions of things that we should be checking to this file if you think there's something missing!!!!***

### CODE REVIEW

**Comment by MathewMacDougall** [comment]
@garethellis0 #122 is merged. Ping me when you've rebased and I'll review.

File: src/thunderbots/software/CMakeLists.txt [hunk 1]
```diff
@@ -250,23 +250,19 @@ if (CATKIN_ENABLE_TESTING)
```
Comment (garethellis0) [inline src/thunderbots/software/CMakeLists.txt:250]: Apologies for a lot of the changes to this file. Ended up reorganizing the files to be alphabetical just to make the merging easier.... but I think it's an overall improvement

File: src/thunderbots/software/test/primitive/primitive.cpp [hunk 2]
```diff
@@ -51,17 +55,79 @@ TEST(PrimitiveTest, create_primitive_from_message_test)
...
+    EXPECT_DOUBLE_EQ(destination.y(), params[1]);
+    EXPECT_DOUBLE_EQ(final_angle.toRadians(), params[2]);
+    EXPECT_DOUBLE_EQ(final_speed, params[3]);
+    EXPECT_EQ(new_prim->getExtraBits(), std::vector<bool>());
+}
+
+// Test that we can correctly translate a ChipPrimitive like:
+// `Primitive` -> `ROS Message` -> `Primitive` and get back the same primitive we
+// started with
+TEST(PrimitiveTest, create_ChipPrimitive_from_message_test)
```
Comment (MathewMacDougall) [inline src/thunderbots/software/test/primitive/primitive.cpp]: test name could be more descriptive, eg. `convert_ChipPrimitive_to_message_and_back_to_ChipPrimitive`. Basically something to indicate we are doing _two_ conversions, not just a single `message -> ChipPrimitive` as the name implies.

File: src/thunderbots/software/test/primitive/primitive.cpp [hunk 3]
```diff
@@ -51,17 +55,79 @@ TEST(PrimitiveTest, create_primitive_from_message_test)
...
+    EXPECT_DOUBLE_EQ(chip_origin.y(), params[1]);
+    EXPECT_DOUBLE_EQ(chip_direction.toRadians(), params[2]);
+    EXPECT_DOUBLE_EQ(chip_distance_meters, params[3]);
+    EXPECT_EQ(new_prim->getExtraBits(), std::vector<bool>());
+}
+
+// Test that we can correctly translate a KickPrimitive like:
+// `Primitive` -> `ROS Message` -> `Primitive` and get back the same primitive we
+// started with
+TEST(PrimitiveTest, create_KickPrimitive_from_message_test)
```
Comment (MathewMacDougall) [inline src/thunderbots/software/test/primitive/primitive.cpp]: ditto

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]
Only small nits to the test names. Can you make a ticket to add similar tests to the other Primitive classes?

**Comment by garethellis0** [comment]
Hmm, I think that tests for the other primitives are in-scope for this PR. Should be fairly small, will add tmw (and address comments).

File: src/thunderbots/software/test/primitive/primitive.cpp (hunk 2 above)
Comment (garethellis0) [reply src/thunderbots/software/test/primitive/primitive.cpp]: Renamed all tests as per the above recommendation.

File: src/thunderbots/software/test/primitive/primitive.cpp (hunk 3 above)
Comment (garethellis0) [reply src/thunderbots/software/test/primitive/primitive.cpp]: Renamed all tests as per the above recommendation.

**Comment by garethellis0** [comment]
Ok, added another test for `CatchPrimitive` to ROS message conversion (and back), and snuck in a few re-namings for variables to make some things in `CatchPrimitive` that I found confusing more clear. Ready for re-review!

**Comment by garethellis0** [comment]
Hmm. I'd like to get this in before the other PR's so that there are examples of the tests people should write. I'm thinking we risk it?

**Review by MathewMacDougall — APPROVED** [review]
YOLO

**Review by jessicahe21 — APPROVED** [review]
LGTM! 👍 

I also believe that it would be helpful to merge this PR first so that there are multiple examples to work with. The conflicts, well, are for the better good as the re-namings does make things clearer and easier to intepret. e.g. `margin` -> `ball_intercept_margin`

As a side note, in `primitive.cpp`, I think what happened was each person wrote a test for their own primitive in the file originally, but they were overwritten during mergings. Thanks for adding the tests back into the file! 😄

**Pull request merged**